src/bridger/Bridger.sol

```diff
@@ -0,0 +1,201 @@
+        // L1GatewayRouter(arbitrumL1GatewayRouter).outboundTransferCustomRefund(
+        //     _signatureData.finalAsset, //token
+        //     REFUND_L2_ACCOUNT, // Account to be credited with the excess gas refund in L2
+        //     _kintoWallet, // Account to be credited with the tokens in L2
+        //     amountBought, // Amount of tokens to bridge
+        //     _swapData.maxGas, // Max gas deducted from user’s L2 balance to cover the execution in L2
+        //     _swapData.gasPriceBid, // Gas price for the execution in L2
+        //     abi.encode(0, bytes("")) // 2 pieces of data encoded: uint256 maxSubmissionCost, bytes extraData
+        // );
+        depositCount++;
+        emit Deposit(msg.sender, _signatureData.inputAsset, _signatureData.amount, _signatureData.finalAsset, amountBought);
+    }
+
+    /* ============ Private methods ============ */
+
+    /**
+     * @dev Permit the spender to spend the specified amount of tokens on behalf of the owner
+     * @param asset address of the token
+     * @param amount amount of tokens
+     * @param expiresAt deadline for the signature
+     * @param signature signature to be recovered
+     */
+    function _permit(address asset, uint256 amount, uint256 expiresAt, bytes memory signature) private {
+        // fix Index range access is only supported for dynamic calldata arrays.
+        (uint8 v, bytes32 r, bytes32 s) = abi.decode(signature, (uint8, bytes32, bytes32));
+        ERC20Permit(asset).permit(msg.sender, address(this), amount, expiresAt, v, r, s);
+    }
+
+    /**
+     * @dev Deposit the specified amount of tokens
+     * @param asset address of the token
+     * @param amount amount of tokens
+     */
+    function _deposit(address asset, uint256 amount) private {
+        require(amount > 0, "Bridger: amount must be greater than 0");
+        require(IERC20(asset).balanceOf(msg.sender) >= amount, "Bridger: insufficient balance");
+        require(IERC20(asset).allowance(msg.sender, address(this)) >= amount, "Bridger: insufficient allowance");
+        IERC20(asset).transferFrom(msg.sender, address(this), amount);
+        deposits[msg.sender][asset] += amount;
+    }
+
+    /**
+     * @dev Swaps ERC20->ERC20 tokens held by this contract using a 0x-API quote.
+     */
+    function _fillQuote(
+        // The `sellTokenAddress` field from the API response.
+        IERC20 sellToken,
+        // The `buyTokenAddress` field from the API response.
+        IERC20 buyToken,
+        // The `allowanceTarget` field from the API response.
+        address spender,
+        // The `to` field from the API response.
+        address payable swapTarget,
+        // The `data` field from the API response.
+        bytes calldata swapCallData
+    ) private returns (uint256) {
+        // Checks that the swapTarget is actually the address of 0x ExchangeProxy
+        // require(swapTarget == exchangeProxy, "Target not ExchangeProxy");
```

Comment:
why did you comment this?

Reply:
because it is not neded in our code but it was in the 0x boilerplate. Ill remove

Reply:
i think we should keep this check

Reply:
given that it requires it to be equal we can hardcode it, no point in keeping the require or creating an extra variable

Reply:
and we can already check that in the lambda/autotask